scio-core/src/main/scala/com/spotify/scio/io/dynamic/package.scala

```diff
@@ -122,6 +129,29 @@ package object dynamic {
 
   }
 
+  implicit class DynamicProtobufSCollection[T <: Message](private val self: SCollection[T])
+      extends AnyVal {
+
+    def saveAsDynamicProtobufFile(
+      path: String,
+      numShards: Int = 0,
+      suffix: String = ".protobuf",
+      codec: CodecFactory = CodecFactory.deflateCodec(6),
+      metadata: Map[String, AnyRef] = Map.empty
+    )(destinationFn: T => String)(implicit cd: ClassTag[T], coder: ScioCoder[T]): Future[Tap[T]] =
+      self.saveAsDynamicAvroFile(
+        path,
+        numShards,
+        AvroBytesUtil.schema,
+        suffix = suffix,
+        codec,
+        metadata
```

Comment:
here we also need to attach `Map("protobuf.generic.schema" -> schema)` to be consistent with `ProtobufIO`